src/transpiler.ts

```diff
@@ -13,7 +13,51 @@ class Transpiler {
             throw new Error('The generated WebAssembly is invalid');
         }
 
-        return new WebAssembly.Instance(new WebAssembly.Module(module.emitBinary()), {}).exports;
+        const wasmModule = new WebAssembly.Module(module.emitBinary());
+
+        return (functionName: string, ...params: any[]) => {
+            let wasmParams = params;
```

Comment:
This is probably not the best name for this variable. Maybe `rewrittenParameters` and `parameters` instead of `params` would be more intention revealing.